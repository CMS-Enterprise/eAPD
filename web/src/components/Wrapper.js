import PropTypes from 'prop-types';
import React, { Component } from 'react';

class Wrapper extends Component {
  componentDidMount() {
    if (!this.props.isDev) return;
    this.addTota11y();
  }

  addTota11y = () => {
    const script = document.createElement('script');
    script.src = '_dev/tota11y.min.js';
    document.body.appendChild(script);
  };

  render() {
    const { children } = this.props;
    return <div className="site">{children}</div>;
  }
}

Wrapper.propTypes = {
  children: PropTypes.node.isRequired,
  isDev: PropTypes.bool.isRequired
};

export default Wrapper;
