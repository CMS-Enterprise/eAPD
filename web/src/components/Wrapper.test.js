import { shallow } from 'enzyme';
import React from 'react';

import Wrapper from './Wrapper';

describe('Wrapper component', () => {
  test('renders correctly in dev mode', () => {
    const component = shallow(<Wrapper isDev>child content</Wrapper>);
    expect(component).toMatchSnapshot();
  });

  test('renders correctly in non-dev mode', () => {
    const component = shallow(<Wrapper isDev={false}>child content</Wrapper>);
    expect(component).toMatchSnapshot();
  });
});
