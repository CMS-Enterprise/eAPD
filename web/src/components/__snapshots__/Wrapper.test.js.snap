// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper component renders correctly in dev mode 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wrapper
    isDev={true}
  >
    child content
  </Wrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "child content",
        <Connect(DataInspector) />,
      ],
      "className": "site",
    },
    "ref": null,
    "rendered": Array [
      "child content",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "child content",
          <Connect(DataInspector) />,
        ],
        "className": "site",
      },
      "ref": null,
      "rendered": Array [
        "child content",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Wrapper component renders correctly in non-dev mode 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wrapper
    isDev={false}
  >
    child content
  </Wrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "child content",
        false,
      ],
      "className": "site",
    },
    "ref": null,
    "rendered": Array [
      "child content",
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "child content",
          false,
        ],
        "className": "site",
      },
      "ref": null,
      "rendered": Array [
        "child content",
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
